pkg/agents/team2/config.go

```diff
@@ -77,58 +73,111 @@ func InitTable(numStates int, numActions int) [][]float64 {
-	qTable := InitTable(27, 3)
+	actionDim := 6
+	daysAtCriticalDim := baseAgent.HealthInfo().MaxDayCritical + 1
+	stateSpace := InitStateSpace(10, 10, daysAtCriticalDim, 11)
+	actionSpace := InitActionSpace(actionDim)
+	policies := InitPolicies(10*10*daysAtCriticalDim*11, actionDim)
+	averagePolicies := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+	rTable := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+	qTable := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+
 	return &CustomAgent2{
-		Base:          baseAgent,
-		stateSpace:    stateSpace,
-		actionSpace:   actionSpace,
-		policies:      policies,
-		rTable:        rTable,
-		qTable:        qTable,
-		isPlatArrived: false,
+		Base:                  baseAgent,
+		stateSpace:            stateSpace,
+		actionSpace:           actionSpace,
+		policies:              policies,
+		averagePolicies:       averagePolicies,
+		rTable:                rTable,
+		qTable:                qTable,
+		visitCounter:          make([]int, 10*10*daysAtCriticalDim*11),
+		daysAtCriticalCounter: 0,
+		PreviousdayAtCritical: 0,
+		lastAge:               0,
+		neiboughHP:            -1,
+		lastAction:            0,
+		lastFoodTaken:         0,
 	}, nil
 }
 
 func (a *CustomAgent2) Run() {
 	//Communication & Observation
 	//communicate before platform arrives to my floor
+	//ask HP of agent below me
+	a.AskNeighbourHP()
+	a.CheckNeighbourHP()
 
-	/*
-		msg := *messages.NewBaseMessage(a.Floor())
-		a.SendMessage(1, msg)
-	*/
-
-	//Perform the following only when platform arrives
-	//NOTE: should let infra team add a func to see whether the plaftfrom has arrived or not
-	if a.CheckState() != -1 {
-		a.isPlatArrived = true
-	} else {
-		a.isPlatArrived = false
-	}
-	if a.isPlatArrived {
+	//Perform the following only once per day when platform arrives
+	//a.exportQTable()
+	//a.exportPolicies()
+
+	if a.PlatformOnFloor() && a.isNewDay() {
 		oldState := a.CheckState()
 		oldHP := a.HP()
 		a.Log("Agent team2 before action:", infra.Fields{"floor": a.Floor(), "hp": oldHP, "food": a.CurrPlatFood(), "state": oldState})
 		action := a.SelectAction()
-		_, err := a.TakeFood(a.actionSpace.actionSet[action](oldHP)) //perform selected action
+
+		foodTaken, err := a.TakeFood(food.FoodType(a.actionSpace[action])) //perform selected action
+		a.lastFoodTaken = foodTaken
 		if err != nil {
-			switch err.(type) {
-			case *infra.FloorError:
-			case *infra.NegFoodError:
-			case *infra.AlreadyEatenError:
-			default:
-			}
+			//if there's error, cease updating tables
+			return
 		}
 		a.Log("Agent team2:", infra.Fields{"selected and performed action": action})
 		a.Log("Agent team2 after action:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "food": a.CurrPlatFood(), "state": a.CheckState()})
+		if a.DaysAtCritical() > 0 && a.PreviousdayAtCritical != a.DaysAtCritical() {
+			a.PreviousdayAtCritical = a.DaysAtCritical()
+			a.daysAtCriticalCounter += 1
+			if a.DaysAtCritical() >= (a.HealthInfo().MaxDayCritical - 1) {
```

Comment:
Will a.DaysAtCritical ever be bigger than MaxDayCritical - 1? (or is this just an extra check?)

Reply:
Yes, MaxDayCritical is the threshold for the agent to enter a critical state. (The variable name could be clearer I agree)